pkg/cmd/server/origin/controller.go

```diff
@@ -3,21 +3,88 @@ package origin
+	kcontroller "k8s.io/kubernetes/pkg/controller"
 	"k8s.io/kubernetes/pkg/serviceaccount"
+	schedulerapi "k8s.io/kubernetes/plugin/pkg/scheduler/api"
+	latestschedulerapi "k8s.io/kubernetes/plugin/pkg/scheduler/api/latest"
 
 	configapi "github.com/openshift/origin/pkg/cmd/server/api"
 	"github.com/openshift/origin/pkg/cmd/server/crypto"
 	"github.com/openshift/origin/pkg/cmd/server/origin/controller"
 )
 
+func (c *MasterConfig) NewKubernetesControllerInitalizers(kc *kubernetes.MasterConfig) (map[string]kubecontroller.InitFunc, error) {
+	ret := map[string]kubecontroller.InitFunc{}
+
+	persistentVolumeController := kubernetes.PersistentVolumeControllerConfig{
+		RecyclerImage: c.ImageFor("recycler"),
+		// TODO: In 3.7 this is renamed to 'Cloud' and is part of kubernetes ControllerContext
+		CloudProvider: kc.CloudProvider,
+	}
+	ret["persistentvolume-binder"] = persistentVolumeController.RunController
+
+	persistentVolumeAttachDetachController := kubernetes.PersistentVolumeAttachDetachControllerConfig{
+		// TODO: In 3.7 this is renamed to 'Cloud' and is part of kubernetes ControllerContext
+		CloudProvider: kc.CloudProvider,
+	}
+	ret["attachdetach"] = persistentVolumeAttachDetachController.RunController
+
+	// TODO: Move this to origin controllers and replace the privileged client with SA
+	// client.
+	var policy *schedulerapi.Policy
+	if _, err := os.Stat(kc.Options.SchedulerConfigFile); err == nil {
+		configData, err := ioutil.ReadFile(kc.SchedulerServer.PolicyConfigFile)
+		if err != nil {
+			return nil, fmt.Errorf("unable to read scheduler config: %v", err)
+		}
+		if err := runtime.DecodeInto(latestschedulerapi.Codec, configData, policy); err != nil {
+			return nil, fmt.Errorf("invalid scheduler configuration: %v", err)
+		}
+	}
+	schedulerController := kubernetes.SchedulerControllerConfig{
+		PrivilegedClient:               kc.KubeClient,
+		SchedulerName:                  kc.SchedulerServer.SchedulerName,
+		HardPodAffinitySymmetricWeight: int(kc.SchedulerServer.HardPodAffinitySymmetricWeight),
+		SchedulerPolicy:                policy,
+	}
+	// FIXME: Move this under openshift controller intialization once we figure out
+	// deployment (options).
+	ret["openshift.io/scheduler"] = schedulerController.RunController
+
+	nodeController := kubernetes.NodeControllerConfig{
+		// TODO: In 3.7 this is renamed to 'Cloud' and is part of kubernetes ControllerContext
+		CloudProvider: kc.CloudProvider,
+	}
+	ret["node"] = nodeController.RunController
+
+	serviceLoadBalancerController := kubernetes.ServiceLoadBalancerControllerConfig{
+		// TODO: In 3.7 this is renamed to 'Cloud' and is part of kubernetes ControllerContext
+		CloudProvider: kc.CloudProvider,
+	}
+	ret["service"] = serviceLoadBalancerController.RunController
+
+	// Add kubernetes controller, the override above takes priority.
+	for name, initFn := range kubecontroller.NewControllerInitializers() {
+		if _, ok := ret[name]; ok {
```

Comment:
skip the resource quota one too.  Add comments explaining that for now its one controller dealing with all types and we must later split it apart.